db/trie/branchroottrie.go

```diff
@@ -138,6 +141,13 @@ func (tr *branchRootTrie) DB() KVStore {
 
 func (tr *branchRootTrie) deleteNodeFromDB(tn Node) error {
 	h := tr.nodeHash(tn)
+	if tr.saveNode {
+		// mark the height-hash to be purged in later pruning
+		tag := byteutil.Uint64ToBytesBigEndian(tr.height)
+		if err := tr.kvStore.Purge(tag, h); err != nil {
+			return err
+		}
+	}
 	return tr.kvStore.Delete(h)
```

Comment:
This is only one call of kvstore.Delete, which means we can catch and save the height-hash in kvStore level. Thus we don't need to modify this file.